source/distributed/shared_tria.cc

```diff
@@ -134,6 +134,16 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      dealii::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
```

Comment:
Ok. I found that in p::Triangulation::copy_triangulation `MPI_Comm_free` is not called. This may cause a memory leak. Am I right?